troposphere/s3.py

```diff
@@ -126,6 +128,27 @@ def validate(self):
                     'properties on S3 Bucket Lifecycle Rule. Please use '
                     '"Transitions" since the former has been deprecated.')
 
+        if 'NoncurrentVersionTransition' in self.properties:
+            if 'NoncurrentVersionTransitions' not in self.properties:
+                # aws moved from a single transition to a list of them
+                # and deprecated 'NoncurrentVersionTransition', so let's
+                # just move it to the new property and not annoy the user.
+                self.properties['NoncurrentVersionTransitions'] = [
```

Comment:
This will always generate the new format even if you use the old standard in py.  So the output will always be compliant.  Given that I don't see are hard wall that someone will hit at some point as the output will always be compliant.

So a warning may generate more noise than positive behaviour.  If we feel that a warning is the right way to go we can look at that.